pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -87,32 +96,25 @@ func (p Provisioner) Provision(options *apibkt.BucketOptions) (*bktv1alpha1.Obje
 		s3svc, err = object.NewS3Agent(p.accessKeyID, p.secretAccessKey, p.getObjectStoreEndpoint(), logger.LevelAt(capnslog.DEBUG), p.tlsCert)
 	}
 	if err != nil {
-		p.deleteOBCResourceLogError("")
 		return nil, err
 	}
 
 	// create the bucket
 	err = s3svc.CreateBucket(p.bucketName)
 	if err != nil {
-		err = errors.Wrapf(err, "error creating bucket %q", p.bucketName)
-		logger.Errorf(err.Error())
-		p.deleteOBCResourceLogError("")
```

Comment:
I remember thinking this was less than ideal to delete the OBCs during failure in every failure path, but where is it being cleaned up now? Or is it not needed with the new idempotent approach?

Reply:
It's not needed with the idempotent approach. L-B-P has generated the bucket name before Provision() for a while, and now with GetUserID being deterministic, there shouldn't be any risk of leaving dangling buckets/users with random names. We can expect to clean up when L-B-P calls Delete().